packages/trpc/server/routers/viewer/teams.tsx

```diff
@@ -735,4 +739,100 @@ export const viewerTeamsRouter = router({
       },
     });
   }),
+  createInvite: authedProcedure
+    .input(
+      z.object({
+        teamId: z.number(),
+      })
+    )
+    .output(z.string())
+    .mutation(async ({ ctx, input }) => {
+      const { teamId } = input;
+      const code = randomBytes(32).toString("hex");
+      await ctx.prisma.invite.create({
+        data: {
+          code,
+          teamId,
+        },
+      });
+      return code;
+    }),
+  setInviteExpiration: authedProcedure
+    .input(
+      z.object({
+        code: z.string(),
+        expireInDays: z.number().optional(),
+      })
+    )
+    .mutation(async ({ ctx, input }) => {
+      const { code, expireInDays } = input;
+
+      const oneDay = 24 * 60 * 60 * 1000;
+      const expiresAt = expireInDays ? new Date(Date.now() + expireInDays * oneDay) : null;
+
+      await ctx.prisma.invite.update({
+        where: { code: code },
+        data: {
+          expiresAt: expiresAt,
+          expireInDays: expireInDays,
+        },
+      });
+    }),
+  deactivateInvite: authedProcedure
+    .input(
+      z.object({
+        code: z.string(),
+      })
+    )
+    .mutation(async ({ ctx, input }) => {
+      const { code } = input;
+      await ctx.prisma.invite.delete({ where: { code } });
+    }),
+  addMemberByLink: authedProcedure
```

Comment:
Why can't we reuse the same procedure (`inviteMember`) to handle both email and code invites? As far as I know both generate a code to accept the invitation. Why do we need another procedure?

Reply:
@zomars If I am not mistaken:

- `inviteMembers` assumes we know who the invitee is. It preemptively creates an membership request for the specified email prior to notifying the user. The link it generates is nothing more than a link to the teams page.

- `addMemberByLink` doesn't have the same assumption. A membership request is created only during the time where the page is accessed given the code.

Reply:
@zomars Would you prefer if we instead seperate `addMemberByLink` logic into two queries? Such that we would have `getInvite` that finds an invite/team based on the code and use `inviteMembers` if any is found?

We would be able to reuse inviteMembers but we would do two queries instead.